TACC-Cloud/hazmapper
# Task/WG-238 panel assets listing

## Overview: ##

This PR adds assets listing and download geojson to assets panel
* adds tree based assets listing
* adds support to delete an feature/asset
* adds support to download geojson (note: not certain if this feature is used)

This PR also:
 * fixes some issues with layout in the map page

## PR Status: ##


## Related Jira tickets: ##

* [WG-238](https://tacc-main.atlassian.net/browse/WG-238)
* [WG-205](https://tacc-main.atlassian.net/browse/WG-205)

## Summary of Changes: ##

## Testing Steps: ##
1. Tests with maps with different types of features
2. Test with maps with long and also short file names
3. Test with maps with data nested in multiple levels
4. Test with duplicate files (this was an issue in angular version)
5. Confirm you can download json file

## UI Photos: ##
![Screenshot 2024-11-06 at 2 52 06 PM](https://github.com/user-attachments/assets/cbbba206-6d3f-4493-b74f-5ba8e845727b)
![Screenshot 2024-11-06 at 2 51 35 PM](https://github.com/user-attachments/assets/b35c3fdc-28e5-4cf3-b6da-1960e9953558)



## Notes: ##

TODO

* [x] fix unit test
* [x] fix linting
* [x] height of entire app on the map project page is off
